app/models/character.rb

```
@@ -26,6 +26,11 @@ class Character < ApplicationRecord
   has_and_belongs_to_many :personality_traits
   has_and_belongs_to_many :archetypes
   has_and_belongs_to_many :locations
+  has_and_belongs_to_many :books
+
+  has_many :protagonist_books, class_name: 'Book', foreign_key: 'protagonist_id'
```

RuboCop (1.69.1)

[convention] Specify a `:dependent` option.

(Rails/HasManyOrHasOneDependent)